lib/src/query_component.dart

```diff
@@ -152,4 +152,36 @@ class QueryComponent implements OnInit {
             InsertionOrder_Budget_BudgetSegment())
         .toList();
   }
+
+  DateTime _getMinStartDate(
+          List<InsertionOrder> insertionOrders) =>
+      insertionOrders
+          .map((io) => Util.convertProtoDateToDateTime(
+              io.budget.activeBudgetSegment.dateRange.startDate))
+          .reduce((date, minDate) => date.isBefore(minDate) ? date : minDate);
+
+  void _addSpentToInsertionOrders(List<InsertionOrder> insertionOrders,
+      Map<String, List<InsertionOrderDailySpend>> spendingMap) {
+    for (final io in insertionOrders) {
+      // Continue if the spending map doesn't have spend data for this io.
+      if (!spendingMap.containsKey(io.insertionOrderId)) continue;
```

Comment:
under what circumstances do we expect this data to be missing?

Reply:
If the IO is active and inflight, but the reporting api doesn't have spending data for it? Sounds like this is not really possible, should we get rid of the sanity check then?

Reply:
if we dont expect this case to happen, I think we should validate the response when we receive it. You can leave this here until we add validation on the service? up to you.